### ISSUE TITLE
Merkaartor Segfaults when double-clicking on virtual node

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2894]
**[Submitted to the original trac issue database at 10.39am, Sunday, 18th April 2010]**

Forwarded from Debian #573237:

Merkaartor Segfaults when double-clicking on virtual node

To reproduce the bug:
[[BR]]
starting merkaartor[[BR]]
activate: use virtual nodes and separate move mode[[BR]]
restart merkaartor[[BR]]
download some ways from osm[[BR]]
use the move tool (from menu bar)[[BR]]
double click a virtual note  -> Segmentation fault[[BR]]

I'll attach a backtrace in a few minutes.

**Comment by tomhughes** [comment]
**Author: bernd@bzed.de**
**[Added to the original trac issue at 10.52am, Sunday, 18th April 2010]**


```
(gdb) bt full
#0  0x00000000004f67de in QString::isEmpty (this=0x3c0000000c0) at /usr/include/qt4/QtCore/qstring.h:694
No locals.
#1  0x00000000005b338f in Feature::id (this=0xf455b60) at Features/Feature.cpp:237
No locals.
#2  0x0000000000642eef in Command::buildUndoList (this=0xc119540, theListWidget=0xfe4f60) at Commands/Command.cpp:67
        it = 0xcad6370
#3  0x0000000000646817 in CommandHistory::buildUndoList (this=0x1405b010, theList=0xfe4f60) at Commands/Command.cpp:439
        i = 2
#4  0x000000000068f881 in DirtyDock::updateList (this=0xd85300) at Docks/DirtyDock.cpp:79
        dirtyObjects = 2
#5  0x0000000000706e99 in DirtyDock::qt_metacall (this=0xd85300, _c=QMetaObject::InvokeMetaMethod, _id=0, _a=0x7fffb345c590) at moc_DirtyDock.cpp:84
No locals.
#6  0x00007f44fa9c4b1f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#7  0x0000000000700b97 in Document::historyChanged (this=0x4394970) at moc_Document.cpp:124
No locals.
#8  0x000000000043018e in Document::addHistory (this=0x4394970, aCommand=0xf3500a0) at Document.cpp:221
No locals.
#9  0x0000000000679419 in MoveNodeInteraction::snapMouseReleaseEvent (this=0x9bc50a0, event=0x7fffb345d2e0, Closer=0x0) at Interactions/MoveNodeInteraction.cpp:200
        Diff = {Lat = 152, Lon = 668}
#10 0x00000000006760ac in FeatureSnapInteraction::mouseReleaseEvent (this=0x9bc50a0, event=0x7fffb345d2e0) at Interactions/Interaction.cpp:219
No locals.
#11 0x00000000004f2d05 in MapView::mouseReleaseEvent (this=0xe68340, event=0x7fffb345d2e0) at MapView.cpp:546
No locals.
#12 0x00007f44fb248be2 in QWidget::event(QEvent*) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#13 0x00000000004f510f in MapView::event (this=0xe68340, event=0x7fffb345d2e0) at MapView.cpp:871
No locals.
#14 0x00007f44fb1f28fc in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#15 0x00007f44fb1f95bb in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#16 0x00007f44fa9b1ddc in QCoreApplication::notifyInternal(QObject*, QEvent*) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#17 0x00007f44fb1f878e in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#18 0x00007f44fb27833b in ?? () from /usr/lib/libQtGui.so.4
No symbol table info available.
#19 0x00007f44fb27704a in QApplication::x11ProcessEvent(_XEvent*) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#20 0x00007f44fb2a2732 in ?? () from /usr/lib/libQtGui.so.4
No symbol table info available.
#21 0x00007f44f501c6c2 in g_main_context_dispatch () from /lib/libglib-2.0.so.0
No symbol table info available.
#22 0x00007f44f5020538 in ?? () from /lib/libglib-2.0.so.0
No symbol table info available.
#23 0x00007f44f50206ec in g_main_context_iteration () from /lib/libglib-2.0.so.0
No symbol table info available.
#24 0x00007f44fa9db233 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#25 0x00007f44fb2a232e in ?? () from /usr/lib/libQtGui.so.4
No symbol table info available.
#26 0x00007f44fa9b0702 in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#27 0x00007f44fa9b0adc in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#28 0x00007f44fa9b481b in QCoreApplication::exec() () from /usr/lib/libQtCore.so.4
No symbol table info available.
#29 0x000000000051fedb in main (argc=1, argv=0x7fffb345e9a8) at Main.cpp:222
        app = <incomplete type>
        logFilename = {static null = {<No data fields>}, static shared_null = {ref = {_q_value = 140798}, alloc = 0, size = 0, data = 0xad4d3a, clean = 0, simpletext = 0, 
            righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, static shared_empty = {ref = {_q_value = 131}, alloc = 0, size = 0, data = 0x7f44facc815a, 
            clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, d = 0xb349c0, static codecForCStrings = 0x0}
        fileNames = {<QList<QString>> = {{p = {static shared_null = {ref = {_q_value = 416034}, alloc = 0, begin = 0, end = 0, sharable = 1, array = {0x0}}, d = 0xad3b20}, 
              d = 0xad3b20}}, <No data fields>}
        args = {<QList<QString>> = {{p = {static shared_null = {ref = {_q_value = 416034}, alloc = 0, begin = 0, end = 0, sharable = 1, array = {0x0}}, d = 0xb55c60}, 
              d = 0xb55c60}}, <No data fields>}
        pixmap = <incomplete type>
        splash = <incomplete type>
        pluginsDir = {d_ptr = {d = 0xbb2730}}
        Main = {<> = {<No data fields>}, <Ui::MainWindow> = {<Ui_MainWindow> = {fileQuitAction = 0xd356b0, helpAboutAction = 0xd2c5f0, fileOpenAction = 0xd33000, 
              viewZoomAllAction = 0xd1ebd0, viewZoomWindowAction = 0xcfb840, viewZoomOutAction = 0xbcd8d0, viewZoomInAction = 0xd4ddf0, createWayAction = 0xb6af30, 
              editUndoAction = 0xd3b410, editRedoAction = 0xd58850, editMoveAction = 0xd279a0, fileImportAction = 0xd45400, fileDownloadAction = 0xbd8f40, 
              createLinearWayAction = 0xd1d6a0, editPropertiesAction = 0xbb6090, fileUploadAction = 0xd221f0, editRemoveAction = 0xcfdaa0, createRoadAction = 0xcfdd10, 
              createNodeAction = 0xcfe180, editReverseAction = 0xcfe1a0, viewGotoAction = 0xcfc530, createDoubleWayAction = 0xcfe700, createRoundaboutAction = 0xcfe9c0, 
              fileNewAction = 0xcfec80, roadSplitAction = 0xcfef40, roadJoinAction = 0xcff200, roadBreakAction = 0xcff7d0, createRelationAction = 0xcff7f0, 
              createAreaAction = 0xcff810, editMapStyleAction = 0xcffad0, mapStyleSaveAction = 0xcffd90, mapStyleLoadAction = 0xcfbd30, createCurvedRoadAction = 0xcfd9c0, 
              toolsPreferencesAction = 0xcfd9e0, exportOSMAllAction = 0xd010d0, exportOSMBinAllAction = 0xd010f0, editSelectAction = 0xd01110, exportOSMViewportAction = 0xd01130, 
              exportOSMBinViewportAction = 0xd01150, bookmarkAddAction = 0xd01800, bookmarkRemoveAction = 0xd01820, nodeMergeAction = 0xd01840, fileSaveAsAction = 0xd021a0, 
              fileSaveAction = 0xd02370, fileDownloadMoreAction = 0xd025a0, action_Docks = 0xd025c0, windowPropertiesAction = 0xd025e0, windowLayersAction = 0xd02600, 
              windowInfoAction = 0xd02620, nodeAlignAction = 0xd028a0, windowDirtyAction = 0xd03250, viewDownloadedAction = 0xd03270, editCopyAction = 0xd03290, 
              editPasteOverwriteAction = 0xd032b0, editPasteMergeAction = 0xd032d0, exportOSMSelectedAction = 0xd03920, exportOSMBinSelectedAction = 0xd03940, 
              editPasteFeaturesAction = 0xd03960, exportOSMAction = 0xd03e60, exportOSMBinAction = 0xd03e80, featureCommitAction = 0xcfb650, exportGPXAction = 0xd004e0, 
              exportKMLAction = 0xd006b0, windowToolbarAction = 0xd05600, windowHideAllAction = 0xd057d0, windowShowAllAction = 0xd059a0, layersAddImageAction = 0xd05b70, 
              renderOsmarenderAction = 0xd05d50, renderNativeAction = 0xd05f30, viewTrackPointsAction = 0xd06100, viewNamesAction = 0xd062e0, gpsConnectAction = 0xd064b0, 
              gpsReplayAction = 0xd06680, windowGPSAction = 0xd06850, gpsDisconnectAction = 0xd06a20, gpsCenterAction = 0xd06bf0, viewTrackSegmentsAction = 0xd06dc0, 
              viewScaleAction = 0xd06fa0, viewRelationsAction = 0xd07170, viewStyleForegroundAction = 0xd07340, viewStyleBackgroundAction = 0xd07520, 
              viewStyleTouchupAction = 0xd07700, gpsRecordAction = 0xd078e0, gpsPauseAction = 0xd07ab0, windowGeoimageAction = 0xd07c80, toolsWorldOsbAction = 0xd07e60, 
              toolsShortcutsAction = 0xd08030, toolTemplatesLoadAction = 0xd08210, toolTemplatesMergeAction = 0xd083f0, toolTemplatesSaveAction = 0xd085d0, 
              relationAddMemberAction = 0xd087b0, relationRemoveMemberAction = 0xd08990, viewArrowsNeverAction = 0xd08b70, viewArrowsOnewayAction = 0xd08d50, 
              viewArrowsAlwaysAction = 0xd08f30, nodeDetachAction = 0xd09110, fileWorkOfflineAction = 0xd09350, renderSVGAction = 0xd09370, windowStylesAction = 0xd09390, 
              toolsWMSServersAction = 0xd093b0, toolsTMSServersAction = 0xd093d0, toolsResetDiscardableAction = 0xd09790, gpsPopupAction = 0xd0a020, cameraAction = 0xd0a250, 
              roadCreateJunctionAction = 0xd0a270, editRotateAction = 0xd0a290, createPolygonAction = 0xd0a2b0, createRectangleAction = 0xd0a2d0, layersNewImageAction = 0xd0a580, 
              editPasteFeatureAction = 0xd0a940, windowFeatsAction = 0xd0a960, roadAddStreetNumbersAction = 0xd0b310, viewVirtualNodesAction = 0xd0b4f0, centralWidget = 0xd0aa90, 
              theMenuBar = 0xd0a650, menuHelp = 0xd0c130, menuCreate = 0xd0c360, menuRoad = 0xd0c010, menuEdit = 0xd0c300, menuView = 0xd0c1d0, menuBookmarks = 0xcfb360, 
              mnuProjections = 0xd0e5d0, menuShow_directional_Arrows = 0xd0dfa0, mnuAreaOpacity = 0xd0f300, menuFile = 0xd0f9b0, menuExport = 0xd100a0, menuRecentOpen = 0xd107c0, 
              menuRecentImport = 0xd0ffd0, menuTools = 0xd11400, menuStyles = 0xd11a00, designerMenu = 0xd12000, menu_Node = 0xd118f0, windowAction = 0xd12c30, menu_Docks = 0xd13240, 
              menu_Feature = 0xd13840, menuLayers = 0xd13e50, menuGps = 0xd14450, menuRelation = 0xd14a50, StatusBar = 0xd15280, toolBar = 0xd5ea10, 
              mobileToolBar = 0xd5f860}, <No data fields>}, <IProgressWindow> = {theProgressDialog = 0x0, theProgressBar = 0xf444cc0, theProgressLabel = 0xf436390}, 
          static staticMetaObject = {d = {superdata = 0xad4860, stringdata = 0x74ba80 "MainWindow", data = 0x74b0e0, extradata = 0x0}}, p = 0xd56390, fileName = {
            static null = {<No data fields>}, static shared_null = {ref = {_q_value = 140798}, alloc = 0, size = 0, data = 0xad4d3a, clean = 0, simpletext = 0, righttoleft = 0, 
              asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, static shared_empty = {ref = {_q_value = 131}, alloc = 0, size = 0, data = 0x7f44facc815a, clean = 0, 
              simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, d = 0x10b4e10, static codecForCStrings = 0x0}, fullscreenState = warning: can't find linker symbol for virtual table for `QByteArray' value
warning:   found `QByteArray::shared_null' instead
{
            static shared_null = {ref = {_q_value = 825}, alloc = 0, size = 0, data = 0xad4d58 "", array = ""}, static shared_empty = {ref = {_q_value = 244}, alloc = 0, size = 0, 
              data = 0x7f44facc8018 "", array = ""}, d = 0x7f44facc8000}, pbImages = 0xe41910, StatusMessage = warning: can't find linker symbol for virtual table for `QString' value
warning:   found `QAbstractButton::staticMetaObject' instead
{static null = {<No data fields>}, static shared_null = {ref = {
                _q_value = 140798}, alloc = 0, size = 0, data = 0xad4d3a, clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, 
            static shared_empty = {ref = {_q_value = 131}, alloc = 0, size = 0, data = 0x7f44facc815a, clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, 
              reserved = 0, array = {0}}, d = 0xad4d20, static codecForCStrings = 0x0}, ViewportStatusLabel = 0xe6d770, PaintTimeLabel = 0xd5e380, theView = 0xe68340, 
          theDocument = 0x4394970, theProperties = 0xe107a0, theInfo = 0xdbd5f0, theDirty = 0xd85300, theLayers = 0xe3f380, theGeoImage = 0x104e8b0, theGPS = 0xdefba0, 
          gpsRecLayer = 0x0, curGpsTrackSegment = 0x0, shortcutsDefault = {{d = 0xe42130, e = 0xe42130}}, qtTranslator = 0xd86a80, merkaartorTranslator = 0xd86840}
        x = 7383760

```

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 2.42pm, Monday, 19th April 2010]**

(In [20936]) FIX : Crash when double-clicking a virtual node in "separate move" mode (fixes #2894)
